File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/error/PlaybackException.java
package com.lasthopesoftware.bluewater.client.playback.file.error;

import com.lasthopesoftware.bluewater.client.playback.file.PlayableFile;

public class PlaybackException extends Exception {
	public final PlayableFile playbackHandler;

	public PlaybackException(PlayableFile playbackHandler) {
		this(playbackHandler, null);
	}

	public PlaybackException(PlayableFile playbackHandler, Throwable cause) {
		super(cause);
		this.playbackHandler = playbackHandler;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/error/PlaybackException.kt
package com.lasthopesoftware.bluewater.client.playback.file.error

import com.lasthopesoftware.bluewater.client.playback.file.PlayableFile

open class PlaybackException @JvmOverloads constructor(
    val playbackHandler: PlayableFile,
    cause: Throwable? = null
) : Exception(cause)
